### ISSUE TITLE
uploadUrl inside uploader function has still default value

### ISSUE THREAD

**Issue by stanislavhannes** [wearebraid/vue-formulate#309]
Hey everyvone,
I have a problem with the file upload. I have defined a global uploader inside my app.js / main.js:
```
...
Vue.use(VueFormulate, {
    library: {
        autocomplete: {
            classification: "text",
            component: AutoComplete,
        },
    },
    uploader: async (file, progress, error, options) => {
        try {
            console.log(options);
            const formData = new FormData();
            formData.append("file", file);
            const result = await fetch(options.uploadUrl, {
                method: "POST",
                body: formData,
            });
            progress(100); // (native fetch doesn’t support progress updates)
            return await result.json();
        } catch (err) {
            error("Unable to upload file");
        }
    },
    //uploader: axiosInstance,
    plugins: [de],
    locale: "de",
    classes: {
        label: (context, classes) => {
            return `uk-form-label formulate-input-label--${context.labelPosition}`;
        },
        errors: "uk-list uk-list-square uk-text-danger uk-text-small",
        inputHasErrors: "uk-form-danger",
        help: "uk-text-light uk-text-small",
    },
});
...
```
And my component that allows to upload pdf files, looks like this:
```
<FormulateInput
            type="file"
            :name="`${year}.financial_statement`"
            :label="'Jahresabschluss ' + year"
            help="Bitte wählen Sie eine PDF Datei aus"
            validation="required|mime:application/pdf"
            :upload-url="$route('key-figures.upload')"
            accept=".pdf"
        />
```

If I choose a PDF file to upload, then I alway get an error. It says `Error: TypeError: Object.defineProperty called on non-object`. I think it has something to do that the config.uploadUrl inside inside my global uploader function, has still the default value "false". If I set the endpoint url manually, it uploads the file and my server responds with the right url. But I still get the error and therefore the upload can't complete.

### Reproduction in CodePen:
https://codepen.io/stanislav-hannes/pen/ExyLOXP

**Expected behavior**
uploadUrl should have the value that is passed via props (related to codePen example it should be "foo")

**Device information:**
 - Device:  MacBook Pro
 - OS: OSX 10.15.7
 - Browser Chrome 86

**Comment by stanislavhannes** [comment]
Well I found a workaround. I downgraded to version 2.4.3 und use the axiosInstance as uploader instead of the function. This seems to work without the TypeError.

**Comment by justin-schroeder** [comment]
I'm taking a comprehensive look at issues related to files in 2.5, so i've thrown this into the mix there, I'll drop a note here though if there are any updates.

**Comment by justin-schroeder** [comment]
Thanks for the report @stanislavhannes. You were totally right that local instances werent getting the prop value of `upload-url` in the options. This is now fixed in the `release/2.5.0` branch!
